src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2311,13 +2298,55 @@ protected void HandleIndexOutputsPerDocument(string documentKey, int numberOfOut
             return null;
         }
 
+        private int? _minBatchSize;
+
+        private const int MinMapBatchSize = 128;
+        private const int MinMapReduceBatchSize = 64;
+
+        private int MinBatchSize
+        {
+            get
+            {
+                if (_minBatchSize != null)
+                    return _minBatchSize.Value;
+
+                switch (Type)
+                {
+                    case IndexType.Map:
+                    case IndexType.AutoMap:
+                        _minBatchSize = MinMapBatchSize;
+                        break;
+                    case IndexType.MapReduce:
+                    case IndexType.AutoMapReduce:
+                        _minBatchSize = MinMapReduceBatchSize;
+                        break;
+                    default:
+                        throw new ArgumentException($"Unknown index type {Type}");
+                }
+
+                return _minBatchSize.Value;
+            }
+        }
+
+        public int GetPageSize()
+        {
+            return IsLowMemory() ? MinBatchSize : int.MaxValue;
```

Comment:
The problem is that in this case, we'll not see the notification that we stopped for low memory, maybe `MinBatchSize +1`?

Reply:
done